kvm/src/lib.rs

```diff
@@ -430,16 +463,16 @@ impl VmFd {
         slot: u32,
         memory_size: usize,
     ) -> Result<Vec<u64>> {
-        let memsize_multiplier = 4096 * 64;
-        // Assert that the memory size is a multiple of 128k (which it must be, because the config
-        // is in megabytes).
-        assert!(
-            memory_size % memsize_multiplier == 0,
-            "memory size {} is not a multiple of 128kB",
-            memory_size
-        );
-        // Create a vec with at least one bit per page of guest memory in this slot.
-        let bitmap_size = memory_size / memsize_multiplier;
+        // Compute the length of the bitmap needed for all dirty pages in one memory slot.
+        // One memory page is 4KiB (4096 bits) and KVM_GET_DIRTY_LOG returns one dirty bit for
+        // each page.
+        let page_size = 4 << 10;
+
+        let div_round_up = |dividend, divisor| (dividend + divisor - 1) / divisor;
+        // For ease of access we are saving the bitmap in a u64 vector. We are using ceil to
```

Comment:
From the comment it is not clear that we are using ceil to calculate the capacity of the `u64` vector. I would rephrase it.
```suggestion
        // For ease of access we are saving the bitmap in an `u64` vector. We are calculating its capacity by dividing the total memory to the size of a page. Even if `mem_size` is not a multiple of `page_size * 64` we make sure all dirty pages are counted by using a ceiling approximation.
```